cmd/shutdown.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type closer interface {
+	Close() error
+}
+
+type structForClose struct {
+	components []closer
+}
+
+//GracefulShutdown implements releasing all resouces it got from system, finish all request handling and return responses when service stopping.
+func (closing *structForClose) GracefulShutdown(done chan<- bool) error {
```

Comment:
It should be unexported

Reply:
WDYT about changing implementation from method to function without new type declaration?